packages/postgres-database/tests/test_utils_folders.py

```diff
@@ -127,11 +130,40 @@ def test_role_permissions(
     ],
 )
 def test__requires_permissions(
-    permissions: list[_FolderPermissions] | None, expected: dict[str, bool]
+    permissions: list[_FolderPermissions], expected: dict[str, bool]
 ):
     assert _requires(*permissions) == expected
 
 
+@pytest.fixture
+async def get_product(
+    connection: SAConnection,
+) -> AsyncIterable[Callable[[str], Awaitable[_ProductName]]]:
+    created_products: list[_ProductName] = []
+
+    async def _(name: str) -> _ProductName:
+        assert name != "osparc", f"{name} is reserved! please choose a different one"
+        resultlt: _ProductName | None = await connection.scalar(
+            products.insert()
+            .values(random_product(name=name, group_id=None))
+            .returning(products.c.name)
+        )
+        assert resultlt is not None
+        return resultlt
+
+    yield _
+
+    for product in created_products:
+        await connection.execute(products.delete().where(products.c.name == product))
+
+
+@pytest.fixture
+async def default_product_name(
+    get_product: Callable[[str], Awaitable[_ProductName]]
+) -> _ProductName:
+    return await get_product("test_product")
+
```

Comment:
please check in pytest-simcore if there is not already what you need for these. no need to duplicate. For example [https://github.com/ITISFoundation/osparc-simcore/blob/autoscaling/ebs-backed-label-ec2s/packages/pytest-simcore/build/lib/pytest_simcore/helpers/faker_factories.py](https://github.com/ITISFoundation/osparc-simcore/blob/autoscaling/ebs-backed-label-ec2s/packages/pytest-simcore/build/lib/pytest_simcore/helpers/faker_factories.py) -> random_product and also [https://github.com/ITISFoundation/osparc-simcore/blob/autoscaling/ebs-backed-label-ec2s/packages/pytest-simcore/build/lib/pytest_simcore/faker_products_data.py](https://github.com/ITISFoundation/osparc-simcore/blob/autoscaling/ebs-backed-label-ec2s/packages/pytest-simcore/build/lib/pytest_simcore/faker_products_data.py)

Reply:
there is also [https://github.com/ITISFoundation/osparc-simcore/blob/5b2d540b192e5ebf2f4615dda21119d4acbbc14b/packages/pytest-simcore/src/pytest_simcore/db_entries_mocks.py](https://github.com/ITISFoundation/osparc-simcore/blob/5b2d540b192e5ebf2f4615dda21119d4acbbc14b/packages/pytest-simcore/src/pytest_simcore/db_entries_mocks.py) where you could do the insertion in the DB.

I believe there are many places where we insert products. Let's reduce duplication and have only one that fail.

Reply:
Also get_product is more like create_product

Reply:
I cannot use the `product` fixture from the above modules. Too many missing dependencies.
I have rested as much as possible form those modules.

Reply:
also renamed fixture